backend/api/serializers/ComplianceReport.py

```diff
@@ -262,12 +266,29 @@ def get_display_name(self, obj):
         return obj.generated_nickname
 
     def get_group_id(self, obj):
-        user = self.context['request'].user
+        user = cache.get("user")
+        if user is None:
+            user = self.context['request'].user
+            cache.set("user", user, 60 * 3)
         return obj.group_id(filter_drafts=user.is_government_user)
 
     def get_supplemental_reports(self, obj):
-        qs = obj.supplemental_reports.order_by('create_timestamp')
-        gov_org = Organization.objects.get(type=1)
+        distinct_supplements_ids = ComplianceReport.objects.filter(
+            latest_report_id=obj.latest_report.id,
+            supplements_id__gt=obj.id
+        ).values('supplements_id').distinct()
+
+        qs = ComplianceReport.objects.filter(
+            Q(id__in=Subquery(distinct_supplements_ids)) | Q(id=obj.latest_report.id)
+        ).order_by('create_timestamp').select_related('type').select_related('status')
+
+        if obj.latest_report.id == obj.id:
+            qs = qs.filter(~Q(id=obj.latest_report.id))
+            
+        gov_org = cache.get("organization-type-1")
```

Comment:
This is a good use of caching because it can be globally available and reused, avoiding the db call each time. We can use this principle elsewhere in the app for more caching situations.